client/web/src/components/Datum/Datum.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import Aux from '../../hoc/Aux/Aux';
+
+import styles from './Datum.module.css';
+
+const datum = (props) => {
+    let datum = null;
+
+    switch (props.type) {
+        case ('passwords'):
+            datum = (
+                <Aux>
+                    <h4>{props.datum.url}</h4>
+                    <p>{props.datum.username}</p>
+                </Aux>
+            );
+            break;
+        case ('creditcards'):
+            datum = (
+                <Aux>
+                    <h4>{props.datum.desc}</h4>
+                    <p>Ending with {props.datum.number.slice(props.datum.number.length - 4)}</p>
+                </Aux>
+            );
+            break;
+        case ('notes'):
+            datum = (
+                <Aux>
+                    <h4>{props.datum.title}</h4>
+                </Aux>
+            );
+            break;
+        case ('codes'):
+            datum = (
+                <Aux>
+                    <h4>{props.datum.desc}</h4>
+                    <p>{props.datum.type}</p>
+                </Aux>
+            );
+            break;
+        default:
+            datum = null;
+            break;
```

Comment:
The code ends up switching over the types of datums in a lot of places. Maybe an enum or some sort of inheritance would be better?